config/v0.3.0/config.go

```diff
@@ -103,10 +111,16 @@ func NewSDK(opts ...ConfigurationOption) (SDK, error) {
 		return noopSDK, err
 	}
 
+	propagator, err := propagator(o)
+	if err != nil {
+		return noopSDK, err
```

Comment:
We need shutdown the providers that were already created so that we do not leak any resources (e.g. hanging goroutines).

Reply:
Alright, I'll move it to the top then, as it's simpler.